lib/components/Input/wheel-picker/base/index.tsx

```diff
@@ -42,41 +64,88 @@ const GenericWheelPickerContent = forwardRef<
             label,
             leftIcon,
             rightIcon,
+            wheelType = "Generic",
+            is12Hour = true,
+            startTimeIn24Format,
+            selectedTime: inputTime = "00:00",
+            WheelPickerContainer = WheelPicker,
+            containerHeight = "100%",
+            hoursInterval = 1,
+            minutesInterval = 1,
             ...rest
-        },
-        ref,
+        }: Omit<ConditionContent<T>, "inputValues" | "setInputValues">,
+        ref: React.ForwardedRef<HTMLDivElement>,
     ) => {
         const containerRef = useRef<HTMLDivElement>(null);
         const actionSheetRef = useRef<HTMLDivElement>(null);
 
         const [inputValues, setInputValues] =
             useState<(string | number)[]>(values);
         const [isPressed, setIsPressed] = useState(false);
+        const [selectedTime, setSelectedTime] = useState<string>(inputTime);
 
-        const initialValues = values.reduce(
-            (previousValue, currentValue, index) => {
-                if (index === 0) return currentValue;
-                return `${previousValue ?? ""} ${currentValue ?? ""}`;
-            },
-            "",
-        ) as string;
+        let initialValues;
 
         const updateItemAtIndex = (
             index: number,
             newValue: string | number,
         ) => {
-            setInputValues((prevItems) => {
-                const updatedItems = [...prevItems];
-                updatedItems[index] = newValue;
-                return updatedItems;
-            });
+            if (wheelType === "Generic")
+                setInputValues((prevItems) => {
+                    const updatedItems = [...prevItems];
+                    updatedItems[index] = newValue;
+                    return updatedItems;
+                });
         };
 
         const { isOpen, open, close, selectedValue, setSelectedValue } =
             useDropdown([containerRef, actionSheetRef]);
 
         const { isMobile } = useBreakpoints();
 
+        if (wheelType === "Generic") {
+            initialValues = values.reduce(
+                (previousValue, currentValue, index): string => {
+                    if (!data?.[index]) return previousValue as string;
+                    if (index === 0) {
+                        const selectedItem = data?.[index].find(
+                            (item) => item.value === currentValue,
+                        );
+
+                        return (reactNodeToString(selectedItem?.label) ||
+                            selectedItem?.value) as string;
+                    } else {
+                        const selectedItem = data?.[index].find(
+                            (item) => item.value === currentValue,
+                        );
+
+                        return `${previousValue ?? ""} ${reactNodeToString(selectedItem?.label) || selectedItem?.value}`;
+                    }
```

Comment:
```suggestion
                    const selectedItem = data?.[index].find(
                        (item) => item.value === currentValue,
                    );
                    if (index === 0) {
                        return (reactNodeToString(selectedItem?.label) ||
                            selectedItem?.value) as string;
                    }
                    return `${previousValue ?? ""} ${reactNodeToString(selectedItem?.label) || selectedItem?.value}`;
```
nit-pick: `selectedItem` is reused and can be declared just once, and `else` can be omitted after `return` statement :)